src/analyses/threadAnalysis.ml

```diff
@@ -22,7 +22,7 @@ struct
   module P = IdentityP (D)
 
   (* transfer functions *)
-  let return ctx (exp:exp option) (f:fundec) : D.t =
+  let return ctx (exp:exp option) _ : D.t =
     let tid = ThreadId.get_current (Analyses.ask_of_ctx ctx) in
     begin match tid with
       | `Lifted tid -> ctx.sideg tid (false, TS.bot (), not (D.is_empty ctx.local))
```

Comment:
Shouldn't this be conditioned on whether the `return` is actually a thread return? Currently any nested function call in a thread would make the entire thread be marked as dirtily exited.

Reply:
The `ThreadAnalysis.return` now queries whether the current return is a thread-return. I also added a test case for the memleak analysis that fails without this check.